Update to 0.10.1

hadess opened the pull request.

hadess committed:
Update to 0.10.1

Ian2020 commented:
Have a go with this:

[generated-sources.json](https://github.com/user-attachments/files/19568867/generated-sources.json)

I tried to check it but ran into another issue:

```
...
npm verbose title npm run package
npm verbose argv "run" "--loglevel" "verbose" "package"
npm verbose logfile logs-max:10 dir:/run/build/pocket-casts/flatpak-node/npm-cache/_logs/2025-04-02T13_29_12_232Z-
npm verbose logfile /run/build/pocket-casts/flatpak-node/npm-cache/_logs/2025-04-02T13_29_12_232Z-debug-0.log
npm verbose stack Error: Missing script: "package"
```

I guess the build process has changed slightly and the `package` script is gone from `package.json`?

hadess committed:
Update release appdata from upstream to 0.10.1

hadess commented:
> Have a go with this:
> 
> [generated-sources.json](https://github.com/user-attachments/files/19568867/generated-sources.json)

For a bizarre reason, the node-gyp SHA was wrong in that file...

> I tried to check it but ran into another issue:

I honestly couldn't figure it out, even looking at the circleci definitions upstream:
```
Running: npx electron-forge make -- --arch=x64,arm64
✔ Checking your system

electron-forge:check-system checking system, create ~/.skip-forge-system-check to stop doing this +0ms
❯ Loading configuration
  electron-forge:project-resolver searching for project in: /run/build/pocket-casts +0ms
  electron-forge:project-resolver electron-forge compatible package.json found in /run/build/pocket-casts/package.json +2ms
✔ Loading configuration
❯ Resolving make targets
✖ Resolving make targets [FAILED: Could not find any make targets configured for the "linux" platform.]

An unhandled rejection has occurred inside Forge:
Error: Could not find any make targets configured for the "linux" platform.
at /run/build/pocket-casts/node_modules/@electron-forge/core/dist/api/make.js:114:27
    at _Task.taskFn (/run/build/pocket-casts/node_modules/@electron-forge/tracer/dist/index.js:58:42)
    at _Task.run (/run/build/pocket-casts/node_modules/listr2/dist/index.cjs:2063:35)
Error: module pocket-casts: Child process exited with code 1
```

Maybe @FelicianoTech has an idea...

I've updated the branch to include the updated `generated-sources.json` file, and my build changes.

Ian2020 commented:
Hi,

I managed to get it to build:

- The SHA was right - the sha is calculated on the URL I think, not the actual file which is why it's probably not what you expect. You'll need to restore to my original generated-sources.
- In `tech.feliciano.pocket-casts.yml`
  - Change the build cmd line 74 to `npx electron-forge package -- --arch=x64,arm64` i.e use package not make. This is what `npm package` was doing originally.
  - The icon has moved: on line 79 use `install -Dm644 img/icon-x512.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png`
- Upstream is now using their own `forge.config.js`, which will conflict/ignore our needed changes to the same config in `package.json`. I think it best to patch `forge.config.js` directly now. I changed the contents of `0001-Avoid-network-access-to-get-Electron-runtime-checksu.patch` to be the following (maybe this file needs renaming also now?):

```
diff --git a/forge.config.js b/forge.config.js
index 468fc4d..5500fc1 100644
--- a/forge.config.js
+++ b/forge.config.js
@@ -1,5 +1,12 @@
 module.exports = {
-	packagerConfig: {},
+	packagerConfig: {
+    "download": {
+      "checksums": {
+        "electron-v34.0.1-linux-x64.zip": "a80804d6469feeb71f8853cdc699a64dc07b07085c4a4c56eedfb24a3049f221",
+        "electron-v34.0.1-linux-arm64.zip": "db16bcca04ee7082f044d75e5410cb450725cb55c5498fac15c6f92079e856d0"
+      }
+    }
+  },
 	makers: [
 		{
 			name: '@electron-forge/maker-zip'
```

This worked for me and I confirmed it runs too.

hadess commented:
> * The SHA was right - the sha is calculated on the URL I think, not the actual file which is why it's probably not what you expect. You'll need to restore to my original generated-sources.

It wasn't me that complained, it was flatpak-builder...

```
Downloading https://registry.npmjs.org/node-gyp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 61196    0 61196    0     0   584k      0 --:--:-- --:--:-- --:--:--  591k
Failed to download sources: module pocket-casts: Wrong sha512 checksum for node-gyp, expected "0e93799501e2164b703479f2e01b55c345d7d5cb93737adc13635a34814bf5299fb20134de3c32fa05806419cd47779646d6c71dd6a11547e1e6388aba30f4d3", was "496ba5fb46b1e2491d971d2cf33a8277a10ffd5b45c75ad4aed506b2077bb4e871538b0b4608a76c9c3d3fce624b2e41503ea88285fec397f5c073d6a93370bd"
```

I could reproduce this after nuking the local `.flatpak-builder` file. I managed to get the app to build after changing that. I've added it as a separate fixup patch in the code I just pushed, if you want to double-check.

> * In `tech.feliciano.pocket-casts.yml`
>   
>   * Change the build cmd line 74 to `npx electron-forge package -- --arch=x64,arm64` i.e use package not make. This is what `npm package` was doing originally.

Done

>   * The icon has moved: on line 79 use `install -Dm644 img/icon-x512.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png`

Done.

> * Upstream is now using their own `forge.config.js`, which will conflict/ignore our needed changes to the same config in `package.json`. I think it best to patch `forge.config.js` directly now. I changed the contents of `0001-Avoid-network-access-to-get-Electron-runtime-checksu.patch` to be the following (maybe this file needs renaming also now?):

I re-did the patch with your changes.

Thank you so much for your help getting this working!

Ian2020 commented:
> It wasn't me that complained, it was flatpak-builder

Oh I see, sorry I misunderstood as I wasn't seeing it locally. I clear my cache and get the same as you now.

I think what's happening is that the URL `https://registry.npmjs.org/node-gyp` is always changing. It probably updates every time a new version of node-gyp is published as it's just a bunch of JSON data about the package. That means this flatpak build and any similar would break every time a new node-gyp is published as the sha changes. That's not good. I'll have a think about what can be done in the generator about that.

hadess commented:
> > It wasn't me that complained, it was flatpak-builder
> 
> Oh I see, sorry I misunderstood as I wasn't seeing it locally. I clear my cache and get the same as you now.
> 
> I think what's happening is that the URL `https://registry.npmjs.org/node-gyp` is always changing. It probably updates every time a new version of node-gyp is published as it's just a bunch of JSON data about the package. That means this flatpak build and any similar would break every time a new node-gyp is published as the sha changes. That's not good. I'll have a think about what can be done in the generator about that.

Alright, that means that we can proceed with that change. I've squashed the fixup and will merge once the build is done. Thanks!
